src/main/java/com/aol/cyclops2/internal/stream/spliterators/DistinctKeySpliterator.java

```diff
@@ -34,7 +34,7 @@ public void forEachRemaining(Consumer<? super T> action) {
         final Consumer<? super IN> toUse = apply(action);
         source.forEachRemaining(e->{
 
-            if(!values.add(keyExtractor.apply(e))){
+            if(values.add(keyExtractor.apply(e))){
```

Comment:
So what was happening here was like so, as it was a negative condition it was adding the item if it was returning false. So the following combination would end up getting 2 values added as being distinct:

["java","java","c","c"]

So first "java" would enter condition if(!values.add(keyExtractor.apply(e))) and not get added, second "java" would get added and the same case with "c".

So basically if you just have 1 instance of the element in the list it never gets added, If you have 2 or more they will get added, If you have 3 elements, for example ["Java","Java","Java"]  you would end up with 2 duplicates